addons/binding/org.openhab.binding.seneye/src/main/java/org/openhab/binding/seneye/handler/SeneyeHandler.java

```diff
@@ -0,0 +1,135 @@
+                newState(readings);
+            } catch (InvalidConfigurationException invalidConfigurationException) {
+                updateStatus(ThingStatus.OFFLINE);
+            }
+        } else {
+            logger.debug("Command {} is not supported for channel: {}", command, channelUID.getId());
+        }
+    }
+
+    @Override
+    public void newState(SeneyeDeviceReading readings) {
+        updateState(CHANNEL_TEMPERATURE, new DecimalType(readings.temperature.curr));
+        updateState(CHANNEL_NH3, new DecimalType(readings.nh3.curr));
+        updateState(CHANNEL_NH4, new DecimalType(readings.nh4.curr));
+        updateState(CHANNEL_O2, new DecimalType(readings.o2.curr));
+        updateState(CHANNEL_PAR, new DecimalType(readings.par.curr));
+        updateState(CHANNEL_PH, new DecimalType(readings.ph.curr));
+        updateState(CHANNEL_LUX, new DecimalType(readings.lux.curr));
+        updateState(CHANNEL_KELVIN, new DecimalType(readings.kelvin.curr));
+        updateState(CHANNEL_LASTREADING, new DateTimeType(readings.status.getLast_experimentDate()));
+        updateState(CHANNEL_SLIDEEXPIRES, new DateTimeType(readings.status.getSlide_expiresDate()));
+    }
+
+    @Override
+    public void invalidConfig() {
+        updateStatus(ThingStatus.OFFLINE);
+    }
+
+    @Override
+    public void initialize() {
+        Configuration conf = this.getConfig();
+
+        String param_aquariumname = String.valueOf(conf.get(PARAMETER_AQUARIUMNAME));
+        String param_username = String.valueOf(conf.get(PARAMETER_USERNAME));
+        String param_password = String.valueOf(conf.get(PARAMETER_PASSWORD));
+        String param_polltime = String.valueOf(conf.get(PARAMETER_POLLTIME));
+
+        if (param_aquariumname.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Check configuration, aquarium name must be provided");
+            return;
+        }
+        if (param_username.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Check configuration, Seneye username must be provided");
+            return;
+        }
+        if (param_password.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Check configuration, Seneye password must be provided");
+            return;
+        }
+        if (param_polltime.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Check configuration, polling time must be provided");
+            return;
+        }
+
+        int polltime = Integer.parseInt(param_polltime) * 60000;
+
+        logger.debug("Initialize Network handler.");
+        this.seneyeService = new SeneyeService(param_aquariumname, param_username, param_password, polltime);
+
+        super.initialize();
+
+        if (seneyeService.initialize()) {

```

Comment:
The `initialize()` method of the handler should return fast, so you should not do http requests within this thread. Please schedule a job for this and set the thing to `unknown`. The scheduled job can set it to `online` once it is finished.

Reply:
indeed, it's a blocking call, but the state of the thing should remain initializing until it's done ?

Reply:
Yes, you should not change the state of the thing in `initialize` if it is not ready. Instead just schedule your job which will set it to ONLINE once the http request was successful.